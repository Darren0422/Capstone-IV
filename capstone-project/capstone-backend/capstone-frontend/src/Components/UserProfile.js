import React from "react";
import UserDetails from "../Modules/UserDetails";
import RepoDetails from "../Modules/RepoDetails";

function UserProfile() {
  // Function to navigate back to the search page
  const goBackToSearch = () => {
    window.location.href = "/";
  };

  return (
    <div>
      <div className="App-api-container">
        <div className="App-api-details">
          <button onClick={goBackToSearch}>Back To Search</button>
          <h2>User Details:</h2>
          <div className="App-api-UserDetails">
            <UserDetails />
          </div>
          <h2>Repo Details:</h2>
          <div className="App-api-RepoDetails">
            <RepoDetails />
          </div>
        </div>
      </div>
    </div>
  );
}

export default UserProfile;

