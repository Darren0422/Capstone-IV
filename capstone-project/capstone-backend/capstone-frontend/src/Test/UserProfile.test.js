import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import UserProfile from '../Components/UserProfile';

// Snapshot test
test('UserProfile component renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <UserProfile />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
