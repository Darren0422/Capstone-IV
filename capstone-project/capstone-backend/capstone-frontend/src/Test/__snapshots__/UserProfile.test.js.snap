// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component renders correctly 1`] = `
<div>
  <div
    className="App-api-container"
  >
    <div
      className="App-api-details"
    >
      <button
        onClick={[Function]}
      >
        Back To Search
      </button>
      <h2>
        User Details:
      </h2>
      <div
        className="App-api-UserDetails"
      >
        <div>
          <p>
            Loading user data...
          </p>
        </div>
      </div>
      <h2>
        Repo Details:
      </h2>
      <div
        className="App-api-RepoDetails"
      >
        <div>
          <p>
            Loading user data...
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
